tests/draft2019-09/ref.json

```diff
@@ -75,13 +75,21 @@
     {
         "description": "escaped pointer ref",
         "schema": {
-            "tilda~field": {"type": "integer"},
-            "slash/field": {"type": "integer"},
-            "percent%field": {"type": "integer"},
+            "$defs": {
```

Comment:
(Just randomly scrolling to this first) -- this change at least as far as I know isn't strictly necessary -- it's not something that's changed in 2019-09 -- would love to keep the changes here strictly minimal to those required for new 2019-09 functionality (or changed 2019-09 functionality).

Reply:
are you referring to the whitespace churn? My editor gets carried away, I'll back it out.

Or the nesting within $defs ? That actually _is_ required, because per 8.2.4.4 it's unspecified what happens when you place or reference schemas under unknown keywords. A conformant implementation may choose to ignore or error (as the one I'm testing does).

Related slack thread: https://json-schema.slack.com/archives/C5CF75URH/p1581043429149800

Reply:
Got it, thanks -- I skimmed that and it does seem to say you're correct for 2019-09, but we still need to be a bit careful, I'm fairly sure this isn't true in earlier drafts, but this change would affect all earlier drafts.

(Worse, if this *is* true in earlier drafts, it'd be incorrect, not just an unintended change, since in earlier drafts it's called something else).

Reply:
Hmm. Unpacking, I think you're pointing out that changing `subSchemas.json` as I have here is a problem for tests of older drafts (good point!).

Changes specific to this `draft2019-09` test hierarchy should only impact implementations of that draft (correct?).

If all that's true, how about forking `subSchemas.json` into `subSchemas-defs.json` as was already been done w/ `name-defs.json`? `subSchemas.json` would be reverted and (only) `draft2019-09` tests would use `subSchemas-defs.json`.

Reply:
Sounds reasonable to me!

Reply:
And sorry for basically typing these out in 30 second increments in between running in and out of meetings :/

Reply:
Oh -- one other thing -- based on what you discussed, it sounds like this is still optional behavior -- so rather than losing the test (i.e. reworking it), we should likely duplicate at least one test of this form in the `optional` directory (i.e. referencing random section of instance).

Reply:
Done and done. Reduced the optional test down to it's essence (the use of remote schemas is irrelevant). PTAL!

Reply:
Awesome, I'll probably pull this/these changes immediately so we can focus on the rest.